### PR TITLE
fix(duckplayer-page): support more external links

### PR DESCRIPTION
**Pull request by shakyShane**
https://app.asana.com/0/0/1206263093562639/f

### CODE REVIEW

**Comment by shakyShane** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #873** <a href="https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/873?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #874** <a href="https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/874?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/873?utm_source=stack-comment).

File: packages/special-pages/pages/duckplayer/src/js/index.js [hunk 1]
```diff
@@ -86,10 +87,42 @@ const VideoPlayer = {
...
+     * In certain circumstances, we may want to intercept
+     * clicks within the iframe - for example when showing a video
+     * that cannot be played in the embed
+     *
+     * @param {string} urlBase - macos/windows current use a different base URL
+     */
+    setClickListener: (urlBase) => {
+        VideoPlayer.onIframeLoaded(() => {
+            const iframe = VideoPlayer.iframe()
+            iframe.contentDocument?.addEventListener('click', (e) => {
```
Comment (jonathanKingston) [inline packages/special-pages/pages/duckplayer/src/js/index.js:109]: What about touch etc?

File: packages/special-pages/pages/duckplayer/src/js/index.js [hunk 2]
```diff
@@ -123,8 +156,15 @@ const VideoPlayer = {
     onIframeLoaded: (callback) => {
         const iframe = VideoPlayer.iframe()
 
-        if (iframe) {
-            iframe.addEventListener('load', callback)
+        if (VideoPlayer.loaded) {
+            callback()
+        } else {
+            if (iframe && !VideoPlayer.loaded) {
```
Comment (jonathanKingston) [inline packages/special-pages/pages/duckplayer/src/js/index.js:162]: Surplus check based on the if conditional.
```suggestion
            if (iframe) {
```

File: packages/special-pages/pages/duckplayer/src/js/index.js (hunk 1 above)
Comment (shakyShane) [reply packages/special-pages/pages/duckplayer/src/js/index.js:109]: Oh that's a good point - I'll fire this up on my windows machine and verify 👍🏻

File: packages/special-pages/pages/duckplayer/src/js/index.js (hunk 2 above)
Comment (shakyShane) [reply packages/special-pages/pages/duckplayer/src/js/index.js]: nice, thanks!

**Review by jonathanKingston — APPROVED** [review]

File: packages/special-pages/pages/duckplayer/src/js/index.js [hunk 3]
```diff
@@ -86,10 +87,42 @@ const VideoPlayer = {
      * Sets up the video player:
      * 1. Fetches the video id
      * 2. If the video id is correctly formatted, it loads the YouTube video in the iframe, otherwise displays an error message
+     * @param {object} opts
+     * @param {string} opts.base
      */
-    init: () => {
+    init: (opts) => {
```
Comment (jonathanKingston) [inline packages/special-pages/pages/duckplayer/src/js/index.js:93]: This singleton pattern is a little odd at this point. We're holding state in an instance, it'd be much simpler to be a class you construct with the state.

Obviously not blocking on this at all, just wanted to state that this is becoming a bit of a smell if we're going to do more in this direction.

Comment (shakyShane) [reply packages/special-pages/pages/duckplayer/src/js/index.js:93]: Agreed yeah - it's from the original port and luckily isolated to this page for now. but we should improve for sure :)

**Pull request merged**